File: backend/app/models/story.py
from sqlalchemy import Column, String, Text, DateTime, Integer
from sqlalchemy.sql import func
from app.config.database import Base

class Story(Base):
    __tablename__ = 'stories'

    id = Column(Integer, primary_key=True, autoincrement=True, index=True)
    title = Column(String(100), nullable=False)
    surface = Column(Text, nullable=False)
    truth = Column(Text, nullable=False)
    tags = Column(Text, nullable=False)  # 存储为JSON字符串
    difficulty = Column(String(20), nullable=False)
    keywords = Column(Text, nullable=False)  # 存储为JSON字符串
    created_at = Column(DateTime(timezone=True), server_default=func.now())

File: frontend/lib/models/story.dart
class Story {
  final int? id;
  final String title;
  final String surface;
  final String truth;
  final List<String> tags;
  final String difficulty;
  final List<String> keywords;

  Story({
    this.id,
    required this.title,
    required this.surface,
    required this.truth,
    required this.tags,
    required this.difficulty,
    required this.keywords,
  });

  factory Story.fromJson(Map<String, dynamic> json) {
    return Story(
      id: json['id'] is int ? json['id'] : int.tryParse(json['id'].toString()),
      title: json['title'],
      surface: json['surface'],
      truth: json['truth'],
      tags: List<String>.from(json['tags']),
      difficulty: json['difficulty'],
      keywords: List<String>.from(json['keywords']),
    );
  }
}
